src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,951 @@
+                var parameterType = invocationSemanticModel.GetTypeInfo(expression, cancellationToken).ConvertedType ?? invocationSemanticModel.Compilation.ObjectType;
+                var parameter = generator.ParameterDeclaration(parameterName, generator.TypeExpression(parameterType));
+                editor.InsertParameter(containingMethod, insertionIndex, parameter);
+            }
+
+            return editor.GetChangedRoot();
+
+            // Adds an argument which is an invocation of the newly created method to the callsites
+            // of the method invocations where a parameter was added.
+            // Example:
+            // public void M(int x, int y)
+            // {
+            //     int f = [|x * y|];
+            //     Console.WriteLine(f);
+            // }
+            // 
+            // public void InvokeMethod()
+            // {
+            //     M(5, 6);
+            // }
+            //
+            // ---------------------------------------------------->
+            // 
+            // public int GetF(int x, int y)
+            // {
+            //     return x * y;
+            // }
+            // 
+            // public void M(int x, int y)
+            // {
+            //     int f = x * y;
+            //     Console.WriteLine(f);
+            // }
+            //
+            // public void InvokeMethod()
+            // {
+            //     M(5, 6, GetF(5, 6)); // This is the generated invocation which is a new argument at the call site
+            // }
+            static SyntaxNode GenerateNewArgumentListSyntaxForTrampoline(ISyntaxFactsService syntaxFacts, ISemanticFactsService semanticFacts,
+                SemanticModel invocationSemanticModel, SyntaxGenerator generator, SyntaxNode currentArgumentListSyntax, SyntaxNode argumentListSyntax,
+                ImmutableArray<IParameterSymbol> validParameters, string newMethodIdentifier, int insertionIndex, CancellationToken cancellationToken)
+            {
+                var invocationArguments = syntaxFacts.GetArgumentsOfArgumentList(argumentListSyntax);
+                var parameterToArgumentMap = MapParameterToArgumentsAtInvocation(semanticFacts, invocationArguments, invocationSemanticModel, cancellationToken);
+                var currentInvocationArguments = syntaxFacts.GetArgumentsOfArgumentList(currentArgumentListSyntax);
+                var requiredArguments = new List<SyntaxNode>();
+
+                foreach (var parameterSymbol in validParameters)
+                {
+                    if (parameterToArgumentMap.TryGetValue(parameterSymbol, out var index))
+                    {
+                        requiredArguments.Add(currentInvocationArguments[index]);
+                    }
+                }
+
+                var methodName = generator.IdentifierName(newMethodIdentifier);
+                var newMethodInvocation = generator.InvocationExpression(methodName, requiredArguments);
```

Comment:
this does not look sufficient. t his will only work if hte callsite has a naked call to the original method.  you need to preserve the expression teh original call is being made off of.  this could `this/base/type-name/aribtrary-expr/etc.`

Reply:
I'll msg you about this